velox/functions/iceberg/tests/TruncateTest.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/iceberg/Truncate.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/core/Expressions.h"
+#include "velox/functions/iceberg/tests/IcebergFunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+
+namespace facebook::velox::functions::iceberg {
+namespace {
+class TruncateTest : public facebook::velox::functions::iceberg::test::
+                         IcebergFunctionBaseTest {
+ public:
+  TruncateTest() {
+    options_.parseDecimalAsDouble = false;
+  }
+
+ protected:
+  template <typename T>
+  std::optional<T> truncate(
+      std::optional<int32_t> width,
+      std::optional<T> value) {
+    return evaluateOnce<T>("truncate(c0, c1)", width, value);
+  }
+
+  template <typename T>
+  std::optional<T> truncate(
+      const TypePtr& type,
+      std::optional<int32_t> width,
+      std::optional<T> value) {
+    return evaluateOnce<T>("truncate(c0, c1)", {INTEGER(), type}, width, value);
+  }
+};
+
+TEST_F(TruncateTest, tinyInt) {
+  EXPECT_EQ(truncate<int8_t>(10, 0), 0);
+  EXPECT_EQ(truncate<int8_t>(10, 1), 0);
+  EXPECT_EQ(truncate<int8_t>(10, 5), 0);
+  EXPECT_EQ(truncate<int8_t>(10, 9), 0);
+  EXPECT_EQ(truncate<int8_t>(10, 10), 10);
+  EXPECT_EQ(truncate<int8_t>(10, 11), 10);
+  EXPECT_EQ(truncate<int8_t>(10, -1), -10);
+  EXPECT_EQ(truncate<int8_t>(10, -5), -10);
+  EXPECT_EQ(truncate<int8_t>(10, -10), -10);
+  EXPECT_EQ(truncate<int8_t>(10, -11), -20);
```

Comment:
The doc says "truncate towards zero", but -11 becoming -20 suggests that truncating is not always towards zero. Would you clarify?

Reply:
I don't recall where I copy the document, but it seems wrong, I updated the document just now.